src/gui/csvImport/CsvImportWidget.cpp

```diff
@@ -0,0 +1,289 @@
+﻿/*
+ *  Copyright (C) 2016 Enrico Mariotti <enricomariotti@yahoo.it>
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 or (at your option)
+ *  version 3 of the License.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "CsvImportWidget.h"
+#include "ui_CsvImportWidget.h"
+#include "gui/MessageBox.h"
+
+#include <QFile>
+#include <QFileInfo>
```

Comment:
We don't really have an guideline for the order of imports. Most of the code uses what louib describes, so maybe it's best to stick with it for now. However, I much dislike it. I would much more prefer

- ui and associated .h file
- project includes
- Qt imports
- system imports